### ISSUE TITLE
What other os can run on campfire2?

### ISSUE THREAD

**Issue by Usulyre** [MilkyDeveloper/Campfire2#1]
HI, what other os can run on campfire2?

Does one create a chroot just like in crouton?

Can one run virtualbox in campfire2?

**Comment by MilkyDeveloper** [comment]
Well, Campfire2 works similar to Crouton. Basically, it uses the chroot command, which changes the root directory of linux to the specified one. Therefore, it needs to works within the limitations of the current kernel (in this case, ChromeOS/Linux). So, all linux distros are supported within Campfire2, but, you can also run virtualbox to virtualize other systems. I would recommend Virt Manager, which is a GUI for QEMU. If you are new to virtualization, VirtualBox is a toy compared to QEMU.

**Comment by Usulyre** [comment]
Thanks for your reply.

So i typed sudo startubuntu but nothing happens? How do I install a desktop environment and packages?What are the basic commands for using campfire2? Thanks for your help

**Comment by MilkyDeveloper** [comment]
Well, it's basically Ubuntu, so you want to type something like `apt update; apt install ubuntu-mate-desktop xserver-xephyr` to install the desktop, and then run `DISPLAY=:0 Xephyr -resizeable -fullscreen :1; export DISPLAY=:1` to initialize the Xserver. Then, you could just run `mate-session` to get a fully working desktop. It's basically Ubuntu, so all `apt` commands will work.

**Comment by Usulyre** [comment]
Can one install other desktop environments? I tried to install other ones like cinnamon but couldn't get anything to open that would resemble a desktop environment.

I tried "apt install ubuntu-mate-desktop xserver-xephyr" and then
"DISPLAY=:0 Xephyr -resizeable -fullscreen :1; export DISPLAY=:1" and it opens a blank window.

I tried "mate-session" and it opened part of a desktop environment the bottom menu bar?

What are the commands to delete the chroot and start over? Any other useful commands?

**Comment by MilkyDeveloper** [comment]
Using the `DISPLAY=:0 Xephyr` command is supposed to open a blank window. Think about it like a nested X server or display. Anyways, put the window in fullscreen and do not change the window's size/dimensions. Then, <kbd>ALT</kbd> <kbd>TAB</kbd> into crosh and type in `mate-session`. It probably should work.

You can install other desktop environments, like KDE, Xfce, LXQT, LXDE, Cinnamon, etc. but not Gnome due to the absence of Systemd. If you use another DE, just make sure to do the Xephyr command first, enter fullscreen/maximized on it, and then type in the command to launch the DE.

You can start over by running `sudo rm -rf /usr/local/ubuntu`, which will nuke the `ubuntu` folder, and then by running:
```
crew install xterm xhost libunwind && echo "Downloading Ubuntu RootFS..." && curl --verbose -o /usr/local/ubuntu.tar.xz "https://cloud-images.ubuntu.com/minimal/releases/focal/release-20210625/ubuntu-20.04-minimal-cloudimg-amd64-root.tar.xz" && sudo rm -rf /usr/local/ubuntu; mkdir /usr/local/ubuntu && sudo tar -xvf /usr/local/ubuntu.tar.xz -C /usr/local/ubuntu && sudo rm /usr/local/ubuntu.tar.xz /usr/local/ubuntu/etc/resolv.conf && sudo mkdir /usr/local/ubuntu/root/Downloads && sudo cp /etc/resolv.conf /usr/local/ubuntu/etc/resolv.conf && sudo curl https://raw.githubusercontent.com/MilkyDeveloper/Campfire2/main/startubuntu -o /usr/local/bin/startubuntu && sudo chmod 755 /usr/local/bin/startubuntu && startsommelier && echo "Done! Run sudo startubuntu to start the chroot."
```

There's not many other useful commands. Because it's Ubuntu, the sky is the limit. You should also note that Campfire2 is *already* running a window manager. Try installing `mesa-utils` (`apt install mesa-utils`) and then type in: `export DISPLAY=:0 glxgears`. The window should pop up on the screen.

**Comment by Usulyre** [comment]
I couldn't get any desktop environment to show up on the blank windows when using "DISPLAY=:0 Xephyr -resizeable -fullscreen :1; export DISPLAY=:1"

I tried this and it works to show desktop, but is this ok? Xephyr -ac -resizeable -fullscreen -terminate -reset :1 & DISPLAY=:1"

Also I can't seem to find a way to quit the blank window? If I force close it, cmdline says it still is running and I have to restart chrome os to get it close completely, any way to completely close without having to resort to system restart?

How does one access local files, meaning my chrome os files or a usb drive or sd card from the desktop environment? It doesn't have any directories that seem to point to any local files or usb drives like crouton.

Also, i tried cinnamon desktop and it reports that it has to run in software rendering mode increasing cpu usage, how do I prevent this. I typed in the cmd echo -e "# Fix 3d Acceleration on Kernel 4.4\nexport MESA_LOADER_DRIVER_OVERRIDE=i965" >> ~/.bashrc && source ~/.bashrc" but it doesn't work with cinnamon desktop, but single apps work.

Does sound work in the desktop environment?
